File: apps/api/app/models/brand_intelligence.py
import uuid
from datetime import datetime
from typing import TYPE_CHECKING, Any

from sqlalchemy import DateTime, Float, ForeignKey, Integer, String, Text, UniqueConstraint
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base, TimestampMixin, UUIDPrimaryKeyMixin

if TYPE_CHECKING:
    from app.models.project import Project


class BrandProfile(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_profiles"
    __table_args__ = (UniqueConstraint("project_id", name="uq_brand_profiles_project_id"),)

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    brand_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    website_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    industry: Mapped[str | None] = mapped_column(String(255), nullable=True)
    country: Mapped[str | None] = mapped_column(String(100), nullable=True)
    short_description: Mapped[str | None] = mapped_column(Text, nullable=True)
    story: Mapped[str | None] = mapped_column(Text, nullable=True)
    mission: Mapped[str | None] = mapped_column(Text, nullable=True)
    values: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    differentiators: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_profile")


class BrandVoice(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_voices"
    __table_args__ = (UniqueConstraint("project_id", name="uq_brand_voices_project_id"),)

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    tone: Mapped[str | None] = mapped_column(String(255), nullable=True)
    style_notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    formality_level: Mapped[str | None] = mapped_column(String(50), nullable=True)
    emoji_policy: Mapped[str | None] = mapped_column(String(100), nullable=True)
    words_to_use: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    words_to_avoid: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    examples_good: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    examples_bad: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_voice")


class BrandProductKnowledge(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_product_knowledge"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    name: Mapped[str] = mapped_column(String(500))
    entity_type: Mapped[str] = mapped_column(String(50), default="product")
    shopify_gid: Mapped[str | None] = mapped_column(String(255), nullable=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    ingredients: Mapped[str | None] = mapped_column(Text, nullable=True)
    origin: Mapped[str | None] = mapped_column(String(255), nullable=True)
    production_process: Mapped[str | None] = mapped_column(Text, nullable=True)
    usage_suggestions: Mapped[str | None] = mapped_column(Text, nullable=True)
    conservation: Mapped[str | None] = mapped_column(Text, nullable=True)
    taste_notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    objections: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    faq: Mapped[list[dict[str, Any]] | None] = mapped_column(JSONB, nullable=True)
    claims_allowed: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    claims_forbidden: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    related_products: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    priority: Mapped[str] = mapped_column(String(20), default="medium")

    project: Mapped["Project"] = relationship(back_populates="brand_product_knowledge")


class BrandAudienceInsight(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_audience_insights"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    segment_name: Mapped[str] = mapped_column(String(255))
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    motivations: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    pain_points: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    objections: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    questions: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    buying_triggers: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_audience_insights")


class BrandClaimRule(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_claim_rules"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    rule_type: Mapped[str] = mapped_column(String(50))
    title: Mapped[str] = mapped_column(String(500))
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    examples: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    severity: Mapped[str] = mapped_column(String(20), default="info")

    project: Mapped["Project"] = relationship(back_populates="brand_claim_rules")


class BrandSeoStrategy(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_seo_strategies"
    __table_args__ = (UniqueConstraint("project_id", name="uq_brand_seo_strategies_project_id"),)

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    primary_keywords: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    secondary_keywords: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    keyword_clusters: Mapped[list[dict[str, Any]] | None] = mapped_column(JSONB, nullable=True)
    priority_pages: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    internal_linking_notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    meta_title_pattern: Mapped[str | None] = mapped_column(String(500), nullable=True)
    meta_description_pattern: Mapped[str | None] = mapped_column(String(500), nullable=True)
    url_handle_pattern: Mapped[str | None] = mapped_column(String(500), nullable=True)
    competitors: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_seo_strategy")


class BrandContentPillar(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_content_pillars"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    name: Mapped[str] = mapped_column(String(255))
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    objective: Mapped[str | None] = mapped_column(Text, nullable=True)
    products: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    channels: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    example_topics: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    cta_notes: Mapped[str | None] = mapped_column(Text, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_content_pillars")


class BrandAiGuardrail(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_ai_guardrails"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    title: Mapped[str] = mapped_column(String(500))
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    rule_type: Mapped[str] = mapped_column(String(50))
    applies_to: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_ai_guardrails")


class BrandAsset(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_assets"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    asset_type: Mapped[str] = mapped_column(String(50))
    name: Mapped[str] = mapped_column(String(255))
    value: Mapped[str | None] = mapped_column(Text, nullable=True)
    file_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_assets")


class BrandImportBatch(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_import_batches"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    source_type: Mapped[str] = mapped_column(String(50), default="file_upload")
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    status: Mapped[str] = mapped_column(String(50), default="pending")
    progress_percent: Mapped[int] = mapped_column(Integer, default=0)
    current_step: Mapped[str | None] = mapped_column(String(500), nullable=True)
    total_files: Mapped[int] = mapped_column(Integer, default=0)
    processed_files: Mapped[int] = mapped_column(Integer, default=0)
    total_facts: Mapped[int] = mapped_column(Integer, default=0)
    approved_facts: Mapped[int] = mapped_column(Integer, default=0)
    rejected_facts: Mapped[int] = mapped_column(Integer, default=0)
    needs_review_facts: Mapped[int] = mapped_column(Integer, default=0)
    error_message: Mapped[str | None] = mapped_column(Text, nullable=True)
    warnings: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)
    started_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    completed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_import_batches")
    documents: Mapped[list["BrandSourceDocument"]] = relationship(
        back_populates="batch",
        cascade="all, delete-orphan",
    )
    extracted_facts: Mapped[list["BrandExtractedFact"]] = relationship(
        back_populates="batch",
        cascade="all, delete-orphan",
    )


class BrandSourceDocument(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_source_documents"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    batch_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("brand_import_batches.id", ondelete="SET NULL"),
        nullable=True,
        index=True,
    )
    filename: Mapped[str] = mapped_column(String(500))
    content_type: Mapped[str] = mapped_column(String(255))
    file_size: Mapped[int] = mapped_column(Integer)
    storage_mode: Mapped[str] = mapped_column(String(50), default="text_only")
    extracted_text: Mapped[str | None] = mapped_column(Text, nullable=True)
    document_type: Mapped[str | None] = mapped_column(String(100), nullable=True)
    document_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    extraction_status: Mapped[str] = mapped_column(String(50), default="uploaded")
    extraction_error: Mapped[str | None] = mapped_column(Text, nullable=True)
    processing_order: Mapped[int | None] = mapped_column(Integer, nullable=True)
    progress_percent: Mapped[int] = mapped_column(Integer, default=0)
    current_step: Mapped[str | None] = mapped_column(String(500), nullable=True)
    extracted_facts_count: Mapped[int] = mapped_column(Integer, default=0)
    needs_review_count: Mapped[int] = mapped_column(Integer, default=0)
    approved_count: Mapped[int] = mapped_column(Integer, default=0)
    rejected_count: Mapped[int] = mapped_column(Integer, default=0)
    uploaded_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
    processed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_source_documents")
    batch: Mapped["BrandImportBatch | None"] = relationship(back_populates="documents")
    extracted_facts: Mapped[list["BrandExtractedFact"]] = relationship(
        back_populates="source_document",
        cascade="all, delete-orphan",
    )


class BrandExtractedFact(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "brand_extracted_facts"

    project_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("projects.id", ondelete="CASCADE"),
        index=True,
    )
    source_document_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("brand_source_documents.id", ondelete="SET NULL"),
        nullable=True,
        index=True,
    )
    batch_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("brand_import_batches.id", ondelete="SET NULL"),
        nullable=True,
        index=True,
    )
    target_section: Mapped[str] = mapped_column(String(100))
    target_entity_type: Mapped[str | None] = mapped_column(String(50), nullable=True)
    field_name: Mapped[str | None] = mapped_column(String(100), nullable=True)
    extracted_value: Mapped[Any] = mapped_column(JSONB, nullable=True)
    source_excerpt: Mapped[str | None] = mapped_column(Text, nullable=True)
    confidence: Mapped[float] = mapped_column(Float, default=0.0)
    status: Mapped[str] = mapped_column(String(50), default="suggested")
    ai_reasoning: Mapped[str | None] = mapped_column(Text, nullable=True)
    is_update_suggestion: Mapped[bool] = mapped_column(default=False, server_default="false")
    existing_target_id: Mapped[uuid.UUID | None] = mapped_column(UUID(as_uuid=True), nullable=True)
    update_mode: Mapped[str] = mapped_column(String(50), default="create")
    previous_value: Mapped[Any | None] = mapped_column(JSONB, nullable=True)
    conflict_status: Mapped[str] = mapped_column(String(50), default="none")
    source_created_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    import_round: Mapped[int | None] = mapped_column(Integer, nullable=True)
    reviewed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    project: Mapped["Project"] = relationship(back_populates="brand_extracted_facts")
    batch: Mapped["BrandImportBatch | None"] = relationship(back_populates="extracted_facts")
    source_document: Mapped["BrandSourceDocument | None"] = relationship(
        back_populates="extracted_facts",
    )

File: packages/shared/src/types/brand-intelligence.ts
export interface BrandProfile {
  id: string;
  projectId: string;
  brandName?: string | null;
  websiteUrl?: string | null;
  industry?: string | null;
  country?: string | null;
  shortDescription?: string | null;
  story?: string | null;
  mission?: string | null;
  values?: string[] | null;
  differentiators?: string[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandVoice {
  id: string;
  projectId: string;
  tone?: string | null;
  styleNotes?: string | null;
  formalityLevel?: string | null;
  emojiPolicy?: string | null;
  wordsToUse?: string[] | null;
  wordsToAvoid?: string[] | null;
  examplesGood?: string[] | null;
  examplesBad?: string[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandProductKnowledge {
  id: string;
  projectId: string;
  name: string;
  entityType: "product" | "category";
  shopifyGid?: string | null;
  description?: string | null;
  ingredients?: string | null;
  origin?: string | null;
  productionProcess?: string | null;
  usageSuggestions?: string | null;
  conservation?: string | null;
  tasteNotes?: string | null;
  objections?: string[] | null;
  faq?: Array<{ question?: string; answer?: string }> | null;
  claimsAllowed?: string[] | null;
  claimsForbidden?: string[] | null;
  relatedProducts?: string[] | null;
  priority: "high" | "medium" | "low";
  createdAt: string;
  updatedAt: string;
}

export interface BrandAudienceInsight {
  id: string;
  projectId: string;
  segmentName: string;
  description?: string | null;
  motivations?: string[] | null;
  painPoints?: string[] | null;
  objections?: string[] | null;
  questions?: string[] | null;
  buyingTriggers?: string[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandClaimRule {
  id: string;
  projectId: string;
  ruleType: "allowed" | "forbidden" | "caution" | "disclaimer";
  title: string;
  description?: string | null;
  examples?: string[] | null;
  severity: "critical" | "warning" | "info";
  createdAt: string;
  updatedAt: string;
}

export interface BrandSeoStrategy {
  id: string;
  projectId: string;
  primaryKeywords?: string[] | null;
  secondaryKeywords?: string[] | null;
  keywordClusters?: Array<Record<string, unknown>> | null;
  priorityPages?: string[] | null;
  internalLinkingNotes?: string | null;
  metaTitlePattern?: string | null;
  metaDescriptionPattern?: string | null;
  urlHandlePattern?: string | null;
  competitors?: string[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandContentPillar {
  id: string;
  projectId: string;
  name: string;
  description?: string | null;
  objective?: string | null;
  products?: string[] | null;
  channels?: string[] | null;
  exampleTopics?: string[] | null;
  ctaNotes?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandAiGuardrail {
  id: string;
  projectId: string;
  title: string;
  description?: string | null;
  ruleType: "must" | "must_not" | "caution";
  appliesTo?: string[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandAsset {
  id: string;
  projectId: string;
  assetType: "logo" | "color" | "font" | "image" | "video" | "document" | "other";
  name: string;
  value?: string | null;
  fileUrl?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandKnowledgeScore {
  overallScore: number;
  status: "incomplete" | "developing" | "ready";
  sectionScores: Record<string, number>;
  missingRequired: string[];
  recommendations: string[];
}

export interface BrandSectionStatus {
  key: string;
  label: string;
  complete: boolean;
  score: number;
}

export interface BrandIntelligenceOverview {
  score: BrandKnowledgeScore;
  sections: BrandSectionStatus[];
  hasProfile: boolean;
  hasVoice: boolean;
  productsCount: number;
  audienceCount: number;
  claimsCount: number;
  guardrailsCount: number;
  pillarsCount: number;
  assetsCount: number;
  sourceDocumentsCount?: number;
  pendingFactsCount?: number;
  pendingSectionDraftsCount?: number;
  latestBatchId?: string | null;
}

export type SectionDraftKey =
  | "brand_profile"
  | "voice_tone"
  | "products_categories"
  | "audience"
  | "claims_compliance"
  | "seo_strategy"
  | "content_pillars"
  | "ai_guardrails"
  | "assets";

export type SectionDraftStatus = "draft" | "needs_review" | "approved" | "rejected" | "applied";

export interface SectionDraftWarnings {
  messages?: string[];
  missingInformation?: string[];
}

export interface BrandSectionDraftListItem {
  id: string;
  batchId?: string | null;
  sectionKey: SectionDraftKey;
  title: string;
  summary?: string | null;
  confidence?: number | null;
  status: SectionDraftStatus;
  sourceFactIds?: string[];
  sourceExternalIds?: string[];
  warnings?: SectionDraftWarnings | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandSectionDraft extends BrandSectionDraftListItem {
  projectId: string;
  draftPayload: unknown;
  sourceDocumentIds?: string[];
  sourceExternalIds?: string[];
  aiReasoning?: string | null;
  previousOfficialSnapshot?: unknown;
  approvedAt?: string | null;
  appliedAt?: string | null;
}

export interface BrandSectionDraftSynthesizeResponse {
  batchId: string;
  draftsCreated: number;
  sections: Array<{
    sectionKey: SectionDraftKey;
    status: SectionDraftStatus;
    confidence?: number | null;
  }>;
}

export interface BrandSectionDraftApplyResultItem {
  draftId: string;
  sectionKey: string;
  status: string;
  message: string;
}

export interface BrandSectionDraftApplyResponse {
  applied?: BrandSectionDraftApplyResultItem[];
  skipped?: BrandSectionDraftApplyResultItem[];
  conflicts?: BrandSectionDraftApplyResultItem[];
}

export type DocumentExtractionStatus =
  | "uploaded"
  | "extracting"
  | "extracted"
  | "failed"
  | "reviewed";

export type FactStatus = "suggested" | "approved" | "rejected" | "needs_review";

export type TargetSection =
  | "brand_profile"
  | "voice_tone"
  | "product_knowledge"
  | "category_knowledge"
  | "audience"
  | "claims_compliance"
  | "seo_strategy"
  | "content_pillars"
  | "ai_guardrails"
  | "assets"
  | "unknown";

export interface BrandSourceDocument {
  id: string;
  projectId: string;
  batchId?: string | null;
  filename: string;
  contentType: string;
  fileSize?: number;
  storageMode?: string;
  documentType?: string | null;
  documentSummary?: string | null;
  extractionStatus: DocumentExtractionStatus | string;
  extractionError?: string | null;
  processingOrder?: number | null;
  progressPercent?: number;
  currentStep?: string | null;
  uploadedAt?: string;
  processedAt?: string | null;
  createdAt?: string;
  updatedAt?: string;
}

export interface BrandSourceDocumentUploadItem {
  id: string;
  filename: string;
  status: string;
}

export type ExternalSourceType =
  | "website"
  | "instagram"
  | "facebook"
  | "tiktok"
  | "youtube"
  | "linkedin"
  | "trustpilot"
  | "google_business"
  | "other";

export type ExternalSourceStatus =
  | "pending"
  | "fetching"
  | "fetched"
  | "failed"
  | "skipped";

export interface BrandExternalSourceInput {
  sourceType: ExternalSourceType;
  url: string;
  label?: string | null;
}

export interface BrandExternalSource {
  id: string;
  projectId: string;
  batchId?: string | null;
  sourceType: ExternalSourceType;
  label?: string | null;
  url: string;
  status: ExternalSourceStatus;
  fetchedTitle?: string | null;
  fetchedText?: string | null;
  fetchedSummary?: string | null;
  fetchError?: string | null;
  lastFetchedAt?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandExternalSourcesFormValues {
  brandName: string;
  websiteUrl: string;
  instagramUrl: string;
  facebookUrl: string;
  tiktokUrl: string;
  youtubeUrl: string;
  linkedinUrl: string;
  trustpilotUrl: string;
  googleBusinessUrl: string;
  otherSources: Array<{ label: string; url: string }>;
}

export interface BrandSourceDocumentsUploadResponse {
  batchId: string;
  status: string;
  documents: BrandSourceDocumentUploadItem[];
  externalSources?: BrandExternalSource[];
}

export type ImportBatchStatus =
  | "pending"
  | "uploading"
  | "extracting"
  | "ai_processing"
  | "review_ready"
  | "partially_failed"
  | "completed"
  | "failed";

export type UpdateMode = "create" | "enrich" | "update" | "duplicate_candidate" | "unknown";
export type ConflictStatus = "none" | "possible_conflict" | "confirmed_conflict";

export interface BrandImportBatchDocumentStatus {
  id: string;
  filename: string;
  extractionStatus: string;
  progressPercent: number;
  currentStep?: string | null;
  extractedFactsCount: number;
  extractionError?: string | null;
}

export interface BrandImportBatch {
  id: string;
  projectId: string;
  name?: string | null;
  sourceType: string;
  notes?: string | null;
  status: ImportBatchStatus;
  progressPercent: number;
  currentStep?: string | null;
  totalFiles: number;
  processedFiles: number;
  totalFacts: number;
  approvedFacts: number;
  rejectedFacts: number;
  needsReviewFacts: number;
  errorMessage?: string | null;
  warnings: string[];
  declaredBrandName?: string | null;
  declaredWebsiteUrl?: string | null;
  startedAt?: string | null;
  completedAt?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandImportBatchStatusResponse extends BrandImportBatch {
  documents: BrandImportBatchDocumentStatus[];
  externalSources?: BrandExternalSource[];
}

export interface BrandImportBatchListItem {
  id: string;
  name?: string | null;
  sourceType: string;
  status: ImportBatchStatus;
  progressPercent: number;
  totalFiles: number;
  totalFacts: number;
  needsReviewFacts: number;
  createdAt: string;
  completedAt?: string | null;
}

export interface BrandImportBatchStartResponse {
  batchId: string;
  status: string;
}

export interface BrandExtractedFact {
  id: string;
  projectId: string;
  sourceDocumentId?: string | null;
  sourceExternalId?: string | null;
  batchId?: string | null;
  targetSection: TargetSection;
  targetEntityType?: string | null;
  fieldName?: string | null;
  extractedValue?: unknown;
  sourceExcerpt?: string | null;
  confidence: number;
  status: FactStatus;
  aiReasoning?: string | null;
  isUpdateSuggestion?: boolean;
  existingTargetId?: string | null;
  updateMode?: UpdateMode;
  previousValue?: unknown;
  conflictStatus?: ConflictStatus;
  sourceCreatedAt?: string | null;
  importRound?: number | null;
  reviewedAt?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface BrandExtractBatchResult {
  documentId: string;
  status: string;
  factsCount: number;
  error?: string | null;
}

export interface BrandExtractBatchResponse {
  results: BrandExtractBatchResult[];
}

export interface BrandApplyFactsResultItem {
  factId: string;
  targetSection: string;
  fieldName?: string | null;
  message: string;
}

export interface BrandApplyFactsResponse {
  saved: BrandApplyFactsResultItem[];
  skipped: BrandApplyFactsResultItem[];
  counts: {
    saved: number;
    skipped: number;
    needsReview: number;
    rejected: number;
  };
}

export interface BrandContextBundle {
  profile?: BrandProfile | null;
  voice?: BrandVoice | null;
  products: BrandProductKnowledge[];
  categories: BrandProductKnowledge[];
  audience: BrandAudienceInsight[];
  claims: BrandClaimRule[];
  seoStrategy?: BrandSeoStrategy | null;
  contentPillars: BrandContentPillar[];
  guardrails: BrandAiGuardrail[];
  assets: BrandAsset[];
  knowledgeScore: BrandKnowledgeScore;
}

export type BrandIntelligenceTab =
  | "overview"
  | "wizard"
  | "import"
  | "profile"
  | "voice"
  | "products"
  | "audience"
  | "claims"
  | "seo"
  | "pillars"
  | "guardrails"
  | "assets"
  | "sources";
